ui/src/components/ArbDataTable/index.js

```diff
@@ -0,0 +1,77 @@
+import React, { useState, useRef } from 'react';
+import Table from 'react-bootstrap/Table';
+
+const columns = [
+	'market',
+	'coin',
+	'from',
+	'to',
+	'buy',
+	'sell',
+	'profit'
+];
+
+function ArbDataTable({
+	from,
+	to,
+	amount,
+	market
+}) {
+	const [data, setData] = useState([]);
+    const [error, setError] = useState(null);
+    const intervalRef = useRef(null);
+
+    React.useEffect(() => {
+        if(intervalRef.current) {
+            clearInterval(intervalRef.current);
+        }
+        fetchData(from, to, amount, market);
+        intervalRef.current = setInterval(() => { 
```

Comment:
Since the results from fetchData calls useEffect anyway, don't can we use setTimeout instead?

Reply:
I added the the dependency array in line #32. So only if any of those values changes during the re render, the useEffect hook will be called